st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -464,6 +470,11 @@ def _render_publish_vars(action_node, action_parameters, execution_result,
         context.update(chain_vars)
         context.update({RESULTS_KEY: previous_execution_results})
         context.update({SYSTEM_SCOPE: KeyValueLookup(scope=SYSTEM_SCOPE)})
+        context.update({
+            DATASTORE_PARENT_SCOPE: {
```

Comment:
Looks like this piece is repeated / duplicated multiple time across the code-base.

Perhaps "get_base_datastore_jinja_context" or similar utility function is in order? :P